### DISCUSSION TITLE
Duplicate logs in Security Onion

### DISCUSSION THREAD

**Discussion by xxvs** [Security-Onion-Solutions/securityonion#7382] [Unsupported Versions] [answered]
I'm using a distributed setup, with one heavy node and one manager node. Although I had the exact same issue on a setup with two heavy nodes and one manager node.

It seems that some logs get repeated, sometimes multiple times. The timestamps are (almost) identical, with a couple of milliseconds difference.

![image](https://user-images.githubusercontent.com/63378913/156329667-add7a25e-6483-492d-8f5a-04d5284bb577.png)

Can I prevent this somehow?

**Comment by dougburks** [comment, marked as answer]
If you drill into those 3 alerts, are all of the fields absolutely identical?

**Reply by xxvs** [reply]
Most of them are, yes. 

Here are the differences:

```
event.timestamp: 2022-03-07T07:58:25.780+0000
log.id.id: 1646639905.135699
log.offset: 8570187
rule.firedtimes: 10
soc_id: wdBjY38BFNIKPQ_xEYcJ

event.timestamp: 2022-03-07T07:58:25.783+0000
log.id.id: 1646639905.136692
log.offset: 8571290
rule.firedtimes: 11
soc_id: v9BjY38BFNIKPQ_xEYcH
```

**Reply by dougburks** [reply]
If `log.id.id` is different, then that suggests that they are not actually duplicate logs.

**Reply by xxvs** [reply]
Any idea why this particular alert would show up 3 times in succession in less than 2ms? It's definitely not 3 separate events.

**Reply by dougburks** [reply]
For `log.id.id: 1646639905.135699` and `log.id.id: 1646639905.136692`, what does the `log.full` field show?

**Reply by xxvs** [reply]
For `log.id.id: 1646639905.135699` there are 3 alerts:
```

Mar  7 07:58:25 [redacted] sudo: pam_unix(sudo:session): session opened for user root by (uid=0)
 Mar  7 07:58:25 [redacted] sudo:    root : TTY=unknown ; PWD=/root ; USER=root ; COMMAND=/sbin/tcpdump -i bond0 -ddd not port 443 && not port 8443
Mar  7 07:58:25 [redacted] sudo: pam_unix(sudo:session): session closed for user root
```
For `log.id.id: 1646639905.136692` it's:

`Mar  7 07:58:25 [redacted] sudo:    root : TTY=unknown ; PWD=/root ; USER=root ; COMMAND=/sbin/tcpdump -i bond0 -ddd not port 443 && not port 8443`

**Reply by dougburks** [reply]
I still don't think these are duplicate logs per se. In any case, you could certainly tune your Wazuh rules to decrease or eliminate these alerts:
https://docs.securityonion.net/en/2.3/wazuh.html#modify-existing-rules

**Comment by reeb4x4** [comment]
I had the same issue - duplicaded identical logs.
Just check if you do not have duplicate entrys in Remote Clusters (elastic->Stack Management->Remote Cluster) with the same Address and different name.
